src/drem/modelling/res_build_stock_postcode.py

```diff
@@ -0,0 +1,309 @@
+"""
+This script will create a synthetic residential building stock.
+
+Our methodology is to divide each SA into several building types,
+using the Split-Apply-Combine pandas transform. These ratios will
+then be applied to outputs from EnergyPlus to generate a first-pass
+estimate for residential energy demand in Dublin.
+
+"""
+import geopandas as gpd
+import numpy as np
+import pandas as pd
+
+from prefect import Flow
+from prefect import task
+
+from drem.filepaths import PROCESSED_DIR
+from drem.filepaths import RAW_DIR
+from drem.filepaths import ROUGHWORK_DIR
+
+
+@task
+def _read_sa_parquet(input_filepath: str) -> pd.DataFrame:
+
+    return pd.read_parquet(input_filepath).drop_duplicates()
+
+
+@task
+def _read_sa_geometries(input_filepath: str) -> gpd.GeoDataFrame:
+
+    return gpd.read_parquet(input_filepath)
+
+
+@task
+def _read_csv(input_filepath: str) -> pd.DataFrame:
+
+    return pd.read_csv(input_filepath, encoding="unicode_escape").drop_duplicates()
+
+
+@task
+def _read_energy_csv(input_filepath: str) -> pd.DataFrame:
+
+    return pd.read_csv(input_filepath)
+
+
+@task
+def _extract_ber_dublin(df: pd.DataFrame, on: str, contains: str) -> pd.DataFrame:
+
+    return df[df[on].str.contains(contains)]
+
+
+@task
+def _lowercase_postcode(df: pd.DataFrame, new: str, old: str) -> pd.DataFrame:
+
+    df[new] = df[old].str.lower()
+
+    return df
+
+
+@task
+def _merge_ber_sa(
+    left: pd.DataFrame, right: pd.DataFrame, left_on: str, right_on: str, **kwargs,
+) -> pd.DataFrame:
+
+    return left.merge(right, left_on=left_on, right_on=right_on, **kwargs)
+
+
+@task
+def _split_ber_construction(df: pd.DataFrame, condition: str) -> pd.DataFrame:
+
+    return df.query(condition).drop_duplicates()
+
+
+@task
+def _extract_res(df: pd.DataFrame, on: str, value: str) -> pd.DataFrame:
+
+    return df.loc[df[on] == value].drop_duplicates().reset_index(drop=True)
+
+
+@task
+def _transform_res(df: pd.DataFrame, lon: str, lat: str) -> gpd.GeoDataFrame:
```

Comment:
Again for _transform_res transform it how?  Ideally I should know what a function does just from its name without reading the implementation